src/github/pullRequestManager.ts

```diff
@@ -0,0 +1,183 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { CredentialStore } from './credentials';
+import { Remote } from '../common/remote';
+import { EventType } from '../common/timelineEvent';
+import { GitHubRepository } from './githubRepository';
+import { Repository, UpstreamRef } from '../api/api';
+import { Protocol } from '../common/protocol';
+import { IssueModel } from './issueModel';
+import { FolderPullRequestManager } from './folderPullRequestManager';
+
+export interface ItemsResponseResult<T> {
+	items: T[];
+	hasMorePages: boolean;
+	hasUnsearchedRepositories: boolean;
+}
+
+export class NoGitHubReposError extends Error {
+	constructor(public repository: Repository) {
+		super();
+	}
+
+	get message() {
+		return `${this.repository.rootUri.toString()} has no GitHub remotes`;
+	}
+}
+
+export class DetachedHeadError extends Error {
+	constructor(public repository: Repository) {
+		super();
+	}
+
+	get message() {
+		return `${this.repository.rootUri.toString()} has a detached HEAD (create a branch first)`;
+	}
+}
+
+export class BadUpstreamError extends Error {
+	constructor(
+		public branchName: string,
+		public upstreamRef: UpstreamRef,
+		public problem: string) {
+		super();
+	}
+
+	get message() {
+		const { upstreamRef: { remote, name }, branchName, problem } = this;
+		return `The upstream ref ${remote}/${name} for branch ${branchName} ${problem}.`;
+	}
+}
+
+export const SETTINGS_NAMESPACE = 'githubPullRequests';
+export const REMOTES_SETTING = 'remotes';
+
+export const PRManagerStateContext: string = 'PRManagerStateContext';
+
+export enum PRManagerState {
+	Initializing = 'Initializing',
+	NeedsAuthentication = 'NeedsAuthentication',
+	RepositoriesLoaded = 'RepositoriesLoaded'
+}
+
+export interface PullRequestDefaults {
+	owner: string;
+	repo: string;
+	base: string;
+}
+
+export const NO_MILESTONE: string = 'No Milestone';
+
+export class PullRequestManager implements vscode.Disposable {
```

Comment:
Sorry, this is only tangentially related to these changes, but while we're reorganizing - do you think this would make more sense named something like `RepositoriesManager`? And the other one `FolderRepositoriesManager`? That seems a bit more accurate and focused to me, I'd eventually like to break up some of the folder one

Reply:
Yes, absolutely. Since I have a dependent PR, I will merge both into alexr00/multifolder then rename.